### PR TITLE
Infinispan cross-site replication failover configuration

### LINKED ISSUE #34355: Add Infinispan failover strategy configuration to the application.properties
### Description

When configuring two sites in Infinispan, we can configure failover backup from one cluster to the other.
Right now this is only possible with the hotrod.properties file, and should be added to the application.properties without forgetting the docs on this matter.

https://infinispan.org/docs/stable/titles/xsite/xsite.html#configuring-hot-rod-clients-cross-site_configuring-cross-site-replication

### Implementation ideas

_No response_

### PR DESCRIPTION
**Pull request by karesti**
Closes https://github.com/quarkusio/quarkus/issues/34355

### CODE REVIEW

File: extensions/infinispan-client/deployment/src/test/java/org/quarkus/infinispan/client/deployment/InfinispanConfigurationSetupTest.java [hunk 1]
```diff
@@ -44,6 +44,11 @@ public void infinispanConnectionConfiguration() {
         assertThat(configuration.security().ssl().provider()).isEqualTo("SSL_prov");
         assertThat(configuration.security().ssl().protocol()).isEqualTo("SSL_protocol");
         assertThat(configuration.security().ssl().ciphers()).containsExactlyInAnyOrder("SSL_cipher1", "SSL_cipher2");
+        assertThat(configuration.clusters().size()).isEqualTo(1);
+        assertThat(configuration.clusters().get(0).getClusterName()).isEqualTo("bsite");
+        assertThat(configuration.clusters().get(0).getCluster().get(0).host()).isEqualTo("bsite1");
+        assertThat(configuration.clusters().get(0).getCluster().get(0).port()).isEqualTo(32111);
+        assertThat(configuration.clusters().get(0).getClientIntelligence()).isEqualTo(ClientIntelligence.BASIC);
```
Comment (geoand) [inline extensions/infinispan-client/deployment/src/test/java/org/quarkus/infinispan/client/deployment/InfinispanConfigurationSetupTest.java:51]: Minor complaint: There are nicer ways to write this in AssertJ, but it's not a problem :)

File: extensions/infinispan-client/runtime/src/main/java/io/quarkus/infinispan/client/runtime/InfinispanClientRuntimeConfig.java [hunk 2]
```diff
@@ -180,6 +180,19 @@ public class InfinispanClientRuntimeConfig {
...
+     * Configures cross site replication from the client with the provided configuration.
+     * This allows automatic failover when a site is down, as well as switching manual with methods like:
+     * <code>
+     *    cacheManager.switchToDefaultCluster();
+     *    cacheManager.switchToCluster('clusterName')
+     * </code>
+     * // @formatter:on
+     */
+    @ConfigItem
+    public Map<String, InfinispanClientRuntimeConfig.BackupClusterConfig> backupCluster = new HashMap<>();
```
Comment (geoand) [inline extensions/infinispan-client/runtime/src/main/java/io/quarkus/infinispan/client/runtime/InfinispanClientRuntimeConfig.java:194]: I am pretty sure we don't need to instantiate the map

File: extensions/infinispan-client/deployment/src/test/java/org/quarkus/infinispan/client/deployment/InfinispanConfigurationSetupTest.java (hunk 1 above)
Comment (karesti) [reply extensions/infinispan-client/deployment/src/test/java/org/quarkus/infinispan/client/deployment/InfinispanConfigurationSetupTest.java]: some refactored done

**Comment by karesti** [comment]
@geoand thanks! pushed some changes

**Review by geoand — APPROVED** [review]
Thanks!

**Pull request merged**